### ISSUE TITLE
highlightColor doesn't work on HexagonLayer

### ISSUE THREAD

**Issue by contra** [uber/deck.gl#2468]
![kapture 2018-11-29 at 12 41 28](https://user-images.githubusercontent.com/425716/49240816-2722ab80-f3d4-11e8-8776-0845d3489a77.gif)

`autoHighlight` works great on the hexagon layer, but the color seems to be ignored and the default is used. This would also be a good opportunity to document these options for the hexagon layer. If this seems uncontroversial I'd be happy to submit a PR - let me know.

#### Actual Result

Highlighting is the default `cyan` highlightColor.

#### Expected Result

Highlighting would be black, given the code sample below.

#### Code Example

```js
  new HexagonLayer({
    id: 'test',
	data,
    extruded: true,
    radius: 25,
    autoHighlight: true,
    highlightColor: [ 0, 0, 0 ]
  })
```

#### To Do List 
  - [ ] Add label and assign to milestone
  - [ ] Coding
  - [ ] Test

#### To Do List 
  - [ ] Add label and assign to milestone 
  - [ ] Coding
  - [ ] Doc update
  - [ ] Whats new update 
  - [ ] Test

**Comment by contra** [comment]
Ah - I seem to have found the issue. If opacity is missing from the color it gets ignored silently.

**Comment by 1chandu** [comment]
Thanks for reporting, I will take a look.

**Comment by 1chandu** [comment]
@contra , you should see following warnings from luma and webgl : 
```
luma: Uniform size should be multiples of 4 (3) [0, 0, 0]
uniforms.js:16 WebGL: INVALID_VALUE: uniform4fv: invalid size
```
I  will update the code to append alpha value similar to how we handle any other color values.

**Comment by 1chandu** [comment]
Resolved by #2479
